webauthn/helpers/decode_credential_public_key.py

```diff
@@ -4,7 +4,7 @@
 try:
     from pydantic.v1 import BaseModel
 except ImportError:
-    from pydantic import BaseModel
+    from pydantic import BaseModel  # type: ignore
```

Comment:
Are the `#type: ignore`'s on the v2 imports because mypy, referencing v1 types, would complain about these?

Reply:
The imports in the `except` block are actually "v1" imports.

If we can't import from `pydantic.v1` (which only exists in v2), then we instead try to import from `pydantic` (assuming that v1 is installed).

Because in the repo itself we have v2 installed, I've put the ignores on the "v1" imports because when v2 is installed that's actually importing the new v2 API which conflicts with the old v1 version.

Reply:
Oh right `pydantic.v1` is the new import 🤦‍♂️ 

So would this mean users still using Pydantic v1 in their project wouldn't get mypy-assisted type annotations on the methods in this library if they're still on Pydantic v1? 🤔

Reply:
Hmm, yeah, I think that's probably right (I have no idea how you'd go about supporting that either) :grimacing:

Reply:
Actually, depending on how much of the internal pydantic stuff is being exposed, I suspect it probably won't be _that_ much of a degraded experience for those folks?

Reply:
> So would this mean users still using Pydantic v1 in their project wouldn't get mypy-assisted type annotations on the methods in this library if they're still on Pydantic v1? 🤔

From what I've seen mypy will also complain about a lot of stuff if you have v2 installed and use the v1 imports, at least if the mypy plugin for pydantic is enabled.